system/jlib/jutil.hpp

```diff
@@ -565,5 +566,17 @@ inline bool stdContains(Container&& container, Value &&v)
     return container.end() != std::find(container.begin(), container.end(), std::forward<Value>(v));
 }
 
+
+class jlib_decl COnScopeExit
+{
+    const std::function<void()> &exitFunc;
+public:
+    inline COnScopeExit(const std::function<void()> &_exitFunc) : exitFunc(_exitFunc) { }
+    inline ~COnScopeExit()
+    {
+        exitFunc();
+    }
+};
```

Comment:
@ghalliday - I couldn't see an equivalent of this, it's a frequent pattern, recoded in various places.
Could be a separate PR ?

Reply:
Looks like a useful addition.